app/services/articles/updater.rb

```diff
@@ -17,18 +17,27 @@ def call
 
       # updated edited time only if already published and not edited by an admin
       update_edited_at = article.user == user && article.published
-      attrs = Articles::Attributes.new(article_params, article.user).for_update(update_edited_at: update_edited_at)
+      # TODO: move setting published_at to Articles::Attributes
+      # when publishing (draft => published), and published_at is nil or is in the past, set it to Time.current
+      # except for exported articles
+      publishing = !article.published && article_params[:published]
```

Comment:
Again, too much logic related to modifying params here, I would like to refactor it as one of the next steps.